src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -147,27 +146,72 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
                 typeParameters: GetTypeParameters(refactoringResult.TypeToExtractFrom, extractInterfaceOptions.IncludedMembers),
                 members: CreateInterfaceMembers(extractInterfaceOptions.IncludedMembers));
 
-            var interfaceDocumentId = DocumentId.CreateNewId(refactoringResult.DocumentToExtractFrom.Project.Id, debugName: extractInterfaceOptions.FileName);
+            switch (extractInterfaceOptions.Location)
+            {
+                case ExtractInterfaceOptionsResult.ExtractLocation.NewFile:
```

Comment:
this is where it probably good to refactor generate new file location code and share everything here rather than write similar code again.

Reply:
New file is inherintly different because of the member selection that extract interface has. The only shared bit is potentially the enum, but I hesitate to merge those in case we want something in the future like `ExtractAbove` and `ExtractBelow` for extract interface.